packages/generator/src/edmx-to-vdm/common/enum-type.ts

```diff
@@ -1,24 +1,81 @@
+import { createLogger, unique } from '@sap-cloud-sdk/util';
 import { ServiceNameFormatter } from '../../service-name-formatter';
-import { VdmEnumType } from '../../vdm-types';
+import { VdmEnumMemberType, VdmEnumType } from '../../vdm-types';
 import { EdmxEnumType } from '../../edmx-parser/v4';
 
+const logger = createLogger({
+  package: 'generator',
+  messageContext: 'enum-type'
+});
+
 export function transformEnumTypesBase(
-  edmTypes: EdmxEnumType[],
+  enumTypes: EdmxEnumType[],
   formatter: ServiceNameFormatter
 ): VdmEnumType[] {
-  const formattedTypes = edmTypes.reduce(
-    (formatted, c) => ({
+  const formattedTypes = enumTypes.reduce(
+    (formatted, enumType) => ({
       ...formatted,
-      [c.Name]: formatter.originalToEnumTypeName(c.Name)
+      [enumType.Name]: formatter.originalToEnumTypeName(enumType.Name)
     }),
     {}
   );
-  return edmTypes.map(e => {
+  return enumTypes.map(e => {
+    e.UnderlyingType = e.UnderlyingType || 'Edm.Int32';
     const typeName = formattedTypes[e.Name];
     return {
       originalName: e.Name,
       typeName,
-      members: e.Member.map(m => m.Name)
+      underlyingType: e.UnderlyingType,
+      members: parseMember(e)
     };
   });
 }
+
+function parseMember(edmxEnumType: EdmxEnumType): VdmEnumMemberType[] {
+  validateUniqueness(edmxEnumType);
+  validateUnderlyingType(edmxEnumType);
+
+  if (areAllValuesSet(edmxEnumType)) {
+    return edmxEnumType.Member.map(member => ({
+      name: member.Name,
+      originalValue: member.Value!
+    }));
+  }
+
+  return edmxEnumType.Member.map((member, index) => ({
+    name: member.Name,
+    originalValue: index.toString()
+  }));
+}
+
+function validateUniqueness(edmxEnumType: EdmxEnumType) {
+  if (unique(edmxEnumType.Member).length !== edmxEnumType.Member.length) {
+    logger.warn(
```

Comment:
use strict flag to decide if it should fail or warn

Reply:
Unfortunately, this flag only exists for OpenAPI, I'll add some TODOs, so it can enabled when we align the two generators.